src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -215,6 +218,39 @@ protected void Initialize(StorageEnvironment environment, DocumentDatabase docum
             }
         }
 
+        private void LoadValues()
+        {
+            TransactionOperationContext context;
+            using (_contextPool.AllocateOperationContext(out context))
+            using (var tx = context.OpenWriteTransaction())
+            {
+                var statsTree = tx.InnerTransaction.CreateTree(Schema.StatsTree);
+
+                var priority = statsTree.Read(Schema.PrioritySlice);
+                if (priority == null)
+                    Priority = IndexingPriority.Normal;
+                else
+                    Priority = (IndexingPriority)priority.Reader.ReadLittleEndianInt32();
+
+                var lastQueryingTime = statsTree.Read(Schema.LastQueryingTimeSlice);
+                if (lastQueryingTime == null)
```

Comment:
I don't think we care about this that much.
We can safely say that on startup, all indexes are marked as last queried on db startup.
Otherwise, you are exposed to issues with last query happening 29.9 minutes ago, so you set it, and then delete the index before the db even have a chance of loading.